components/validator/validator.go

```diff
@@ -135,24 +151,30 @@ func (v *Validator) loop() {
 }
 
 func (v *Validator) submitL2Output() error {
-	cCtx, cancel := context.WithTimeout(v.ctx, 3*time.Minute)
-	defer cancel()
-
-	output, shouldSubmit, err := v.l2os.FetchNextOutputInfo(cCtx)
+	ctx := v.ctx
+	output, shouldSubmit, err := v.l2os.FetchNextOutputInfo(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to fetch next output: %w", err)
 	}
 	if !shouldSubmit {
 		return nil
 	}
 
-	tx, err := v.l2os.CreateSubmitL2OutputTx(cCtx, output)
+	data, err := v.l2os.SubmitL2OutputTxData(output)
 	if err != nil {
-		return fmt.Errorf("failed to create submit l2 output transaction: %w", err)
+		return fmt.Errorf("failed to create submit l2 output transaction data: %w", err)
 	}
-	if err := v.SendTransaction(cCtx, tx); err != nil {
+
+	txCandidate := txmgr.TxCandidate{
+		TxData:   data,
+		To:       &v.cfg.L2OutputOracleAddr,
+		GasLimit: 0,
+	}
+
+	if err := v.sendTransaction(ctx, txCandidate); err != nil {
```

Comment:
Here, there is a timeout of 10 minutes in case of Optimism before sending the transaction, but I didn't add it because it didn't exist originally in ours. If this part is missing, I will add it like below. Please check it.
```suggestion
	cCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
	defer cancel()
	if err := v.sendTransaction(ctx, txCandidate); err != nil {
```

Reply:
I think we need this in case there's no response from a server!

Reply:
`sendTransaction`, which is called after the context declaration with 10 minutes timeout, calls `TxManager.Send`. At the beginning of `TxManager.Send`, there is a part to declare timeout context using `TxManager.Config.TxSendTimeout`.
So I thought that the 10 minute timeout before `sendTransaction` is unnecessary, wdyt?

Reply:
I added some env variables to `ops-devnet/docker-compose.yml` to set `TxManager.Config.TxSendTimeout` 10 min.